### ISSUE TITLE
Define model metadata story

### ISSUE THREAD

**Issue by kylehounslow** [Digitalist-USA/tara#4]
In the same directory as trained model, we need to include a JSON file with the following:  
* General info
  * when was it trained
  * who trained it
* Dataset info
  * path(s) to storage buckets used in training.
  * how many images used in training
  * resolution of images used in training
* Training parameters  
  * hyperparameters used
* Metrics
  * accuracy etc

**Comment by mikaellaine** [comment]
Why csv? Why the extra characters of json? Do we foresee more complex data structures?

**Comment by mikaellaine** [comment]
Should this feature be implemented on the model file level, where hyperparameters are used, accuracy is available etc?

**Comment by kylehounslow** [comment]
@mikaellaine because JSON string and Python `dict` play very nice together.  
```
import json
my_dict = {'foo': 'bar', 'learning_rates': [0.01, 0.001, 0.0001]}
json_str = json.dumps(my_dict)  # serialize to json.
my_dict2 = json.loads(json_str)  # deserialize to dict
```  
  
Yes it will be on the model file level. Basically this is a simple way to keep track of our experiments and will likely be automated to save at each model checkpoint. Since the model architecture won't change much over the next few months it is important that we track the above metadata.

**Comment by mikaellaine** [comment]
Fair enough. Should I start defining this and creating a first impl on the tf model?

**Comment by kylehounslow** [comment]
Save .cfg file to job dir solves this problem
